### PR TITLE
Add Swizzling for AppDelegate lifecycle methods

### PR DESCRIPTION
**Pull request by iTOYS**

### CODE REVIEW

File: src/WindowsAzureMessaging/WindowsAzureMessaging/MSNotificationHub.m [hunk 1]
```diff
@@ -36,10 +36,37 @@ + (void)initWithConnectionString:(NSString *) connectionString withHubName:(NSSt
     
 }
 
+- (void)registerForRemoteNotifications {
+#if TARGET_OS_OSX
```
Comment (evgkarpov) [inline src/WindowsAzureMessaging/WindowsAzureMessaging/MSNotificationHub.m:40]: Same, we don't need any targets except iOS

File: src/WindowsAzureMessaging/WindowsAzureMessaging/Internal/MSNotificationHubAppDelegate.m [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    BOOL forwarderEnabled = forwarderEnabledNum ? [forwarderEnabledNum boolValue] : YES;
+    self.enabled = forwarderEnabled;
+    if (self.enabled) {
+        NSLog(@"Delegate forwarder for info.plist key '%@' enabled. It may use swizzling.", plistKey);
+    } else {
+        NSLog(@"Delegate forwarder for info.plist key '%@' disabled. It won't use swizzling.", plistKey);
+    }
+}
+
+- (void) custom_application:(UIApplication *)application didRegisterForRemoteNotificationsWithDeviceToken:(NSData *)deviceToken {
```
[the remark this answers is not in the document]
Comment (iTOYS) [reply src/WindowsAzureMessaging/WindowsAzureMessaging/Internal/MSNotificationHubAppDelegate.m:87]: it's for swizzling

File: src/WindowsAzureMessaging/WindowsAzureMessaging/Internal/MSNotificationHubAppDelegate.m [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+}
+
+- (void)custom_setDelegate:(id <UIApplicationDelegate>)delegate {
+ 
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        [[MSNotificationHubAppDelegate sharedInstance]  swizzleImplForMethod: @selector(application: didRegisterForRemoteNotificationsWithDeviceToken:) inClass:[delegate class] from:[MSNotificationHubAppDelegate class] forDelegate:NO];
+        [[MSNotificationHubAppDelegate sharedInstance]  swizzleImplForMethod: @selector(application: didFailToRegisterForRemoteNotificationsWithError:) inClass:[delegate class] from:[MSNotificationHubAppDelegate class] forDelegate:NO];
+        [[MSNotificationHubAppDelegate sharedInstance]  swizzleImplForMethod: @selector(application: didReceiveRemoteNotification:) inClass:[delegate class] from:[MSNotificationHubAppDelegate class] forDelegate:NO];
+#if !TARGET_OS_OSX
```
Comment (mpodwysocki) [inline src/WindowsAzureMessaging/WindowsAzureMessaging/Internal/MSNotificationHubAppDelegate.m:47]: We are not targeting macOS, so not sure why this is necessary...

File: src/WindowsAzureMessaging/WindowsAzureMessaging/Internal/MSNotificationHubAppDelegate.m [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+
+- (void) custom_application:(UIApplication *)application didFailToRegisterForRemoteNotificationsWithError:(NSError *)error {
+    [MSNotificationHub didFailToRegisterForRemoteNotificationsWithError:error];
+}
+
+- (void) custom_application:(UIApplication *)application didReceiveRemoteNotification:(NSDictionary *)userInfo {
+    [MSNotificationHub didReceiveRemoteNotification:userInfo];
+}
+
+#if !TARGET_OS_OSX
```
Comment (mpodwysocki) [inline src/WindowsAzureMessaging/WindowsAzureMessaging/Internal/MSNotificationHubAppDelegate.m:99]: Once again, not targeting macOS at this point, so not necessary

**Comment by iTOYS** [comment]
@mpodwysocki @evgkarpov all comments resolved, please review again

**Pull request merged**